cosmoz-tree.html

```diff
@@ -22,56 +29,78 @@
 			throw new Error('Must be implemented in derived object');
 		},
 
-		getPath: function (node) {
+		getPathNodes: function (node) {
 			throw new Error('Must be implemented in derived object');
 		},
 
-		getPathByProperty: function (propertyName, propertyValue) {
+		getPathString: function (node) {
 			throw new Error('Must be implemented in derived object');
 		},
 
-		getChildren: function (node) {
+		getChildren: function (node, asList) {
 			throw new Error('Must be implemented in derived object');
 		},
 
 		getProperty: function (node, propertyName) {
 			throw new Error('Must be implemented in derived object');
 		},
 
-		getPathString: function (node, textPropertyName, separator) {
-			var path = this.getPath(node);
-			return path.map(function (n) {
-				return this.getProperty(n, textPropertyName);
-			}, this).join(separator);
-		},
+		/**
+		 * Returns True if a node fullfills the search criteria.
+		 */
+		nodeConformsSearch: function (node, propertyName, propertyValue, exact) {
+			var property = this.getProperty(node, propertyName);
 
-		getPathStringByProperty: function (propertyName, propertyValue, textPropertyName, separator) {
-			var node = this.getNodeByProperty(propertyName, propertyValue);
-			if (node !== undefined) {
-				return this.getPathString(node, textPropertyName, separator);
+			switch (exact) {
+				case true:
+					if (property === propertyValue) {
```

Comment:
I'd say rather just `return property === propertyValue;` and skip the `break;`
Same approach with `case false:` and `default`